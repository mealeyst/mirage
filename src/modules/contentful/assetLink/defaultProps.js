import { default as responsiveProps } from '../responsive/images/defaultProps'

const defaultProps = {
  fields: {
    asset: responsiveProps,
    destination: '/shop/boys/all'
  },
  sys: {
    contentType: {
      sys: {
        id: 'assetLink'
      }
    }
  }
}

export default defaultProps
