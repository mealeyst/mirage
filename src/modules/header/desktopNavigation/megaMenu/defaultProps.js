export const regions = {
  left: [
    {
      id: 'category',
      title: 'Shop by Category',
      links: [
        {target: '#', text: 'Shop New', src: 'https://res.cloudinary.com/roa-canon/web/megamenu/NAV_G_NewArrivals.jpg'},
        {target: '#', text: 'Best Sellers', src: 'https://res.cloudinary.com/roa-canon/web/megamenu/NAV_G_BestSellers.jpg'},
        {target: '#', text: 'Graphics Shop', src: 'https://res.cloudinary.com/roa-canon/web/megamenu/NAV_G_GraphicTees.jpg'},
        {target: '#', text: 'Tees & Tops', src: 'https://res.cloudinary.com/roa-canon/web/megamenu/NAV_G_Tops.jpg'},
        {target: '#', text: 'Sweatshirts', src: 'https://res.cloudinary.com/roa-canon/web/megamenu/NAV_G_Sweats.jpg'},
        {target: '#', text: 'Dresses & Skirts', src: 'https://res.cloudinary.com/roa-canon/web/megamenu/NAV_G_Dresses.jpg'},
        {target: '#', text: 'Leggings & Sweatpants', src: 'https://res.cloudinary.com/roa-canon/web/megamenu/NAV_G_Pants.jpg'},
        {target: '#', text: 'Active', src: 'https://res.cloudinary.com/roa-canon/web/megamenu/NAV_G_Active.jpg'},
        {target: '#', text: 'Outerwear', src: 'https://res.cloudinary.com/roa-canon/web/megamenu/NAV_G_Outerwear.jpg'},
        {target: '#', text: 'Swim', src: 'https://res.cloudinary.com/roa-canon/web/megamenu/NAV_G_Swim.jpg'},
        {target: '#', text: 'Accessories', src: 'https://res.cloudinary.com/roa-canon/web/megamenu/NAV_G_Accessories.jpg'},
        {target: '#', text: 'Shop All', src: 'https://res.cloudinary.com/roa-canon/web/megamenu/NAV_G_All.jpg'}
      ]
    },
  ],
  right: [
    {
      id: 'story',
      title: 'Shop by Story',
      links: [
        {href: '#', text: 'Next-Level Neon', src: 'https://res.cloudinary.com/roa-canon/web/megamenu/NAV_E_Neon.jpg'},
        {href: '#', text: 'Not Your Average Rainbow', src: 'https://res.cloudinary.com/roa-canon/web/megamenu/NAV_E_Rainbow.jpg'},
        {href: '#', text: 'Awesome Graffiti', src: 'https://res.cloudinary.com/roa-canon/web/megamenu/NAV_E_Graffiti.jpg'},
        {href: '#', text: 'Vacation-Ready Styles', src: 'https://res.cloudinary.com/roa-canon/web/megamenu/NAV_E_Vacation.jpg'}
      ]
    },
  ]
}
