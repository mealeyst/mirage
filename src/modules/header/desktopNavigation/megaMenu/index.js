export { default as MegaMenu } from './megaMenu'
export { default as MegaMenuSection } from './megaMenuSection'
