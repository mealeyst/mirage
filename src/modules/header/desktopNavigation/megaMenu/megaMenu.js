import React from 'react'
import PropTypes from 'prop-types'
import styled, { keyframes } from 'styled-components'
import { CSSTransitionGroup } from 'react-transition-group'
import classNames from 'classnames'

import {
  FlexCol,
  FlexRow,
  InlineImage,
  MegaMenuSection,
  NavyLink,
} from 'SRC'


const FadeIn = keyframes`
  0% {
    opacity: 0;
  }
  10% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
`

const FadeOut = keyframes`
  0% {
    opacity: 1;
  }
  10% {
    opacity: 1;
  }
  99% {
    opacity: 0;
    height: auto;
  }
  100% {
    opacity: 0;
    height: 0;
  }
`

export class BaseMegaMenu extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      closed: props.closed,
      closing: props.closing,
      menuImage: undefined
    }
    this.menu = undefined
  }

  componentDidUpdate (prevProps) {
    const { visible: currentlyVisible, animationLength } = this.props
    if (!currentlyVisible && currentlyVisible !== prevProps.visible) {
      this.setState({closing: true})
      setTimeout(() => { this.setState({closed: true})}, animationLength)
    } else if (currentlyVisible && currentlyVisible !== prevProps.visible) {
      this.setState({closed: false, closing: false})
    }
  }

  setRef = (elem) => {
    this.menu = elem
  }
  setMenuImage = (src) => () => {
    this.setState({menuImage: src})
  }

  clearMenuImage = () => {
    this.setState({menuImage: undefined})
  }

  render () {
    const { className, regions, visible } = this.props
    const { menuImage: src, closing, closed } = this.state
    const classes = classNames(className, {
      'opened': visible,
      'closed': closed,
      'closing': closing
    })
    return (
      <div ref={this.setRef} className={classes}>
        <FlexRow element='menu' constrained>
          <FlexCol desktop={{width: 3, span: 2}}>
            {regions.left.map((section, id) => {
              return (
                <MegaMenuSection
                  key={`left-${id}`}
                  section={section}
                  setMenuImage={this.setMenuImage}
                  clearMenuImage={this.clearMenuImage} />
              )
            })}
          </FlexCol>
          <FlexCol desktop={{width: 3}}>
          {regions.right.map((section, id) => {
            return (
              <MegaMenuSection
                key={`right-${id}`}
                section={section}
                setMenuImage={this.setMenuImage}
                clearMenuImage={this.clearMenuImage} />
            )
          })}
          </FlexCol>
          <FlexCol desktop={{width: 4}}>
            <div className='imageContainer'>
              {src && <InlineImage aria-hidden className='menuImage' src={src} />}
            </div>
          </FlexCol>
        </FlexRow>
      </div>
    )
  }
}

const MegaMenu = styled(BaseMegaMenu)`
  overflow: hidden;
  padding-top: 4rem;
  padding-bottom: 4rem;
  background-color: ${props => props.theme.colors.white};
  box-shadow: 0 2px 1px 0 rgba(0,0,0,.16);
  border-top: 0.2rem solid ${props => props.theme.colors.rocketBlue};
  border-bottom: 0.2rem solid ${props => props.theme.colors.rocketBlue};
  z-index: 100;
  &.closed {
    opacity: 0;
    height: 0;
    padding: 0;
  }
  &.closing {
    opacity: 0;
    animation: ${FadeOut} ${props => (props.animationLength / 1000)}s ease-in;
  }
  &.opened {
    opacity: 1;
    animation: ${FadeIn} ${props => (props.animationLength / 1000)}s ease-in;
  }
  ${NavyLink} {
    margin-bottom: 1rem;
    display: inline-block;
  }

    li {
      list-style: none;
    }
  }
  .imageContainer {
    position: relative;
    width: 100%;
    height: 100%;
    display: flex;
  }
  .menuImage {
    position: absolute;
    left: 0;
    top: 0;
    width: auto;
    max-width: 100%;
    max-height: 30rem;
    margin-top: 2rem;
    border: 0.1rem solid ${props => props.theme.colors.rocketBlue};
    border-radius: 0.2rem;
    animation: ${FadeIn} ${props => (props.animationLength / 1000)}s ease-in;
  }
`

MegaMenu.propTypes = {
  animationLength: PropTypes.number,
  className: PropTypes.string,
  closed: PropTypes.bool,
  closing: PropTypes.bool,
  regions: PropTypes.shape({
    left: PropTypes.array,
    right: PropTypes.array
  }),
  visible: PropTypes.bool
}

MegaMenu.defaultProps = {
  closed: true,
  closing: false,
  animationLength: 300,
  visible: false
}

/** @component */
export default MegaMenu
