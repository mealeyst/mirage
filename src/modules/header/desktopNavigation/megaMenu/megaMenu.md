```js
    <MegaMenu
      closed={false}
      closing={false}
      regions={require('./defaultProps').regions}
      visible />
```
