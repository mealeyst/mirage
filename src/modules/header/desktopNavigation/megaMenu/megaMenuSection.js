import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { H4 } from 'SRC'

import { MegaMenuLink } from 'SRC'

class BaseMegaMenuSection extends React.Component {

  render () {
    const {className, section, ...props} = this.props
    if (section) {
      return (
        <ol className={className} key={section.id} id={section.id}>
          {
            (section.title && section.title !== '') &&
            <H4>{section.title}</H4>
          }
          {section.links.map((link, id) => {
            return (
              <li key={`${section}-${id}`}>
                <MegaMenuLink
                  link={link}
                  id={`${section}-${id}`}
                  {...props} />
              </li>
            )
          })}
        </ol>
      )
    } else { return null }
  }
}

const MegaMenuSection = styled(BaseMegaMenuSection)`
  padding-left: 0;
  > li {
    list-style: none;
  }
`

MegaMenuSection.propTypes = {
  className: PropTypes.string,
  clearMenuImage: PropTypes.func.isRequired,
  section: PropTypes.shape({
    title: PropTypes.string,
    links: PropTypes.array
  }).isRequired,
  setMenuImage: PropTypes.func.isRequired,
  rednerLink: PropTypes.func
}

/** @component */
export default MegaMenuSection
