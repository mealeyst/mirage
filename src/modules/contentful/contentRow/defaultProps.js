import { toutDefaultProps } from 'SRC/modules/contentful/touts/defaultProps'
// import { default as responsiveVideoProps } from 'SRC/modules/contentful/responsive/video/defaultProps'

export default {
  sys: {
    id: 'example-content-row'
  },
  fields: {
    layout: 'Full Bleed',
    entryElements: [
      ...toutDefaultProps
    ]
  }
}
