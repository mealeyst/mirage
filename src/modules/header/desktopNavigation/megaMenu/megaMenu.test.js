import React from 'react'
import 'jest-styled-components'

import { MegaMenu, MegaMenuSection } from 'SRC'

import { BaseMegaMenu } from './megaMenu'

import { regions } from './defaultProps'

const { mountWithTheme, shallowWithTheme } = global

describe('(Styled Component) MegaMenu', () => {
  const createMegaMenu = (inProps) => {
    const props = {
      regions: regions,
      ...inProps
    }
    return shallowWithTheme(<MegaMenu {...props} />)
  }

  test('matching the snapshot', () => {
    expect(createMegaMenu())
    .toMatchSnapshot()
  })
})

describe('(Base Component) BaseMegaMenu', () => {
  const createBaseMegaMenu = (inProps) => {
    const props = {
      regions: regions,
      ...inProps
    }
    return mountWithTheme(<BaseMegaMenu {...props} />)
  }

  test('rendering the left and right regions ', () => {
    expect(createBaseMegaMenu().find(MegaMenuSection).length).toEqual(2)
  })

  test('setting the mega menu image on mouseenter', () => {
    const component = createBaseMegaMenu()
    expect(component.state().menuImage).toBeUndefined()
    component.find({children: regions.left[0].links[0].text}).first().simulate('mouseenter')
    expect(component.state().menuImage).toEqual(
      regions.left[0].links[0].src
    )
  })

  test('clearing the mega menu image on mouseenter', () => {
    const component = createBaseMegaMenu()
    expect(component.state().menuImage).toBeUndefined()
    component.find({children: regions.left[0].links[0].text}).first().simulate('mouseenter')
    expect(component.state().menuImage).toEqual(
      regions.left[0].links[0].src
    )
    component.find({children: regions.left[0].links[0].text}).first().simulate('mouseleave')
    expect(component.state().menuImage).toBeUndefined()
  })
})
