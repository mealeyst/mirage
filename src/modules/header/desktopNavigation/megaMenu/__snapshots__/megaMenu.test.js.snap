// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Styled Component) MegaMenu matching the snapshot 1`] = `
.c0 {
  overflow: hidden;
  padding-top: 4rem;
  padding-bottom: 4rem;
  background-color: #FFFFFF;
  box-shadow: 0 2px 1px 0 rgba(0,0,0,.16);
  border-top: 0.2rem solid #0073D1;
  border-bottom: 0.2rem solid #0073D1;
}

.c0.closed {
  opacity: 0;
  height: 0;
  padding: 0;
}

.c0.closing {
  opacity: 0;
  -webkit-animation: hPNQcY 0.3s ease-in;
  animation: hPNQcY 0.3s ease-in;
}

.c0.opened {
  opacity: 1;
  -webkit-animation: cussWI 0.3s ease-in;
  animation: cussWI 0.3s ease-in;
}

.c0 .c1 {
  margin-bottom: 1rem;
  display: inline-block;
}

.c0 li {
  list-style: none;
}

.c0 .imageContainer {
  position: relative;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 .menuImage {
  position: absolute;
  left: 0;
  top: 0;
  width: auto;
  max-width: 100%;
  max-height: 30rem;
  margin-top: 2rem;
  border: 0.1rem solid #0073D1;
  border-radius: 0.2rem;
  -webkit-animation: cussWI 0.3s ease-in;
  animation: cussWI 0.3s ease-in;
}

<BaseMegaMenu
  animationLength={300}
  className="c0"
  closed={true}
  closing={false}
  regions={
    Object {
      "left": Array [
        Object {
          "id": "category",
          "links": Array [
            Object {
              "src": "https://res.cloudinary.com/roa-canon/web/megamenu/NAV_G_NewArrivals.jpg",
              "target": "#",
              "text": "Shop New",
            },
            Object {
              "src": "https://res.cloudinary.com/roa-canon/web/megamenu/NAV_G_BestSellers.jpg",
              "target": "#",
              "text": "Best Sellers",
            },
            Object {
              "src": "https://res.cloudinary.com/roa-canon/web/megamenu/NAV_G_GraphicTees.jpg",
              "target": "#",
              "text": "Graphics Shop",
            },
            Object {
              "src": "https://res.cloudinary.com/roa-canon/web/megamenu/NAV_G_Tops.jpg",
              "target": "#",
              "text": "Tees & Tops",
            },
            Object {
              "src": "https://res.cloudinary.com/roa-canon/web/megamenu/NAV_G_Sweats.jpg",
              "target": "#",
              "text": "Sweatshirts",
            },
            Object {
              "src": "https://res.cloudinary.com/roa-canon/web/megamenu/NAV_G_Dresses.jpg",
              "target": "#",
              "text": "Dresses & Skirts",
            },
            Object {
              "src": "https://res.cloudinary.com/roa-canon/web/megamenu/NAV_G_Pants.jpg",
              "target": "#",
              "text": "Leggings & Sweatpants",
            },
            Object {
              "src": "https://res.cloudinary.com/roa-canon/web/megamenu/NAV_G_Active.jpg",
              "target": "#",
              "text": "Active",
            },
            Object {
              "src": "https://res.cloudinary.com/roa-canon/web/megamenu/NAV_G_Outerwear.jpg",
              "target": "#",
              "text": "Outerwear",
            },
            Object {
              "src": "https://res.cloudinary.com/roa-canon/web/megamenu/NAV_G_Swim.jpg",
              "target": "#",
              "text": "Swim",
            },
            Object {
              "src": "https://res.cloudinary.com/roa-canon/web/megamenu/NAV_G_Accessories.jpg",
              "target": "#",
              "text": "Accessories",
            },
            Object {
              "src": "https://res.cloudinary.com/roa-canon/web/megamenu/NAV_G_All.jpg",
              "target": "#",
              "text": "Shop All",
            },
          ],
          "title": "Shop by Category",
        },
      ],
      "right": Array [
        Object {
          "id": "story",
          "links": Array [
            Object {
              "href": "#",
              "src": "https://res.cloudinary.com/roa-canon/web/megamenu/NAV_E_Neon.jpg",
              "text": "Next-Level Neon",
            },
            Object {
              "href": "#",
              "src": "https://res.cloudinary.com/roa-canon/web/megamenu/NAV_E_Rainbow.jpg",
              "text": "Not Your Average Rainbow",
            },
            Object {
              "href": "#",
              "src": "https://res.cloudinary.com/roa-canon/web/megamenu/NAV_E_Graffiti.jpg",
              "text": "Awesome Graffiti",
            },
            Object {
              "href": "#",
              "src": "https://res.cloudinary.com/roa-canon/web/megamenu/NAV_E_Vacation.jpg",
              "text": "Vacation-Ready Styles",
            },
          ],
          "title": "Shop by Story",
        },
      ],
    }
  }
  visible={false}
/>
`;
